### DISCUSSION TITLE
How to change task icon?

### DISCUSSION THREAD

**Discussion by igormoochnick** [ebullient/obsidian-task-collector#53] [Q&A] [answered]
I get all the icons in the popup as "checkbox". How do I assign different icons for different task symbols?  I do not get the interface that you show on the animated image that you show on your git homepage

**Comment by ebullient** [comment]
In settings: 
<img width="614" alt="image" src="https://user-images.githubusercontent.com/808713/164989091-6aa4b59e-4a62-4565-9a21-a7a5379afbfc.png">

**Comment by igormoochnick** [comment]
I've followed the instructions but then I see this in the "Mark task" popup (BTW: I use "Atom" theme)
<img width="240" alt="image" src="https://user-images.githubusercontent.com/1222369/164996537-a4f1c7e7-4a28-4a61-8f12-9ea330bb86ac.png">

**Reply by ebullient** [reply]
AH! I see what you mean. Ok.. The Atom theme is forcibly styling your checkboxes (probably with svg).  I hate to recommend another plugin, but... 
[Snippetor](https://github.com/ebullient/obsidian-snippetor) was built for this (and you can remove it afterwards).

Add the Snippetor plugin, it will grab your configured checkmarks.. style them the way you want (let me know if the style doesn't apply correctly, it should). Export the snippet, and then use that. ;)

**Comment by igormoochnick** [comment]
Thank you for your fast responses.
I've tried a couple of different other themese, "Obisidian Default", "Deep Thought", etc... and had the same effect.
Installed, activated and configured Snipetor - fun plugin. It produced some result - will take it for a spin and will update you on the progress.
Thank you for all your help!

**Reply by ebullient** [reply]
Do you have any other snippets active? I test snippetor with Obsidian default.. this is what a snippetor checkboxes snippet looks like on the default theme: 
<img width="325" alt="image" src="https://user-images.githubusercontent.com/808713/165011255-58939a06-55c2-42f1-a185-bd16c3bc920c.png">

(your colors and values will differ.. but they should render. ;))

The Task Collector modal is emulating the task list style (so it borrows task mark styling for free). If tasks always have a hard check like that, that means you have something rendering that checkmark as an image (from a snippet or something.. )

**Reply by igormoochnick** [reply]
No snippets whatsoever. Never used them. Didn't even have a "snippets" folder (which was causing the Snipptor to fail at the beginning).
So for testing my theory, I've disabled ALL task-related plugins like (Kanban, Tasks, TQ, etc...), disabled the Snipetor, and tried a bunch of different themes with only TC plugin enabled - no effect. The tasks popup showing up like this:
<img width="345" alt="image" src="https://user-images.githubusercontent.com/1222369/165078225-54c65da4-0a5a-4d2b-8ffe-802193da614d.png">

**Reply by igormoochnick** [reply]
There is also something else going on with the syle - note the icons on the left:
<img width="694" alt="image" src="https://user-images.githubusercontent.com/1222369/165079356-b0920066-1693-4beb-a68e-0db8fa68792f.png">

**Reply by ebullient** [reply]
Can you try the default theme with all other snippets disabled? also.. I recommend that what you specify in the input boxes that are displaying be something easy to type (e.g. `!`) .. if you expand the block (arrows on the right),there will be another box you can use to specify the display character (e.g. an emoji).

**Reply by igormoochnick** [reply]
As I've mentioned above that is exactly what I did: default theme, ALL plugins disabled and NO snippets whatsoever.
<img width="241" alt="image" src="https://user-images.githubusercontent.com/1222369/165413619-a9f392e7-5319-4a95-a62a-52c99fa9c4e3.png">
<img width="198" alt="image" src="https://user-images.githubusercontent.com/1222369/165413654-63ea234a-75ea-417b-9846-ff2d81ddf56b.png">

**Reply by ebullient** [reply]
Ok. Default theme with no other snippets.. the above is what I would expect.
If you now only turn on the snippetor snippet.. then what happens?
(keep the default theme)

**Comment by ebullient** [comment, marked as answer]
This question is effectively a duplicate of #61 

It comes down to the correct application of a theme or style that provides additional task styles.
